plugins/inputs/turbostat/turbostat.go

```diff
@@ -0,0 +1,307 @@
+type fieldMap = map[string]any
+
+func (*Turbostat) SampleConfig() string {
+	return sampleConfig
+}
+
+// Init initialises the plugin.
+func (t *Turbostat) Init() error {
+	if t.Path == "" {
+		t.Path = "turbostat"
+	}
+
+	if t.Interval <= 0 {
+		t.Interval = config.Duration(10 * time.Second)
+	}
+
+	t.command = t.buildCmd()
+	return nil
+}
+
+// Start starts the plugin.
+func (t *Turbostat) Start(acc telegraf.Accumulator) error {
+	var err error
+	t.process, err = process.New(t.command, make([]string, 0))
+	if err != nil {
+		return fmt.Errorf("failed to create process %s: %w", t.command, err)
+	}
+
+	t.process.ReadStderrFn = func(r io.Reader) {
+		if err := processStderr(r, acc); err != nil {
+			acc.AddError(err)
+		}
+	}
+	t.process.ReadStdoutFn = func(r io.Reader) {
+		if err := processStdout(r, acc); err != nil {
+			acc.AddError(err)
+		}
+	}
+	t.process.StopOnError = false
+	t.process.Log = t.Log
+
+	if err = t.process.Start(); err != nil {
+		return fmt.Errorf("failed to start process %s: %w", t.command, err)
+	}
+	return nil
+}
+
+// Stop stops the plugin.
+func (t *Turbostat) Stop() {
+	t.process.Stop()
+}
+
+func (*Turbostat) Gather(telegraf.Accumulator) error {
+	return nil
+}
+
+// buildCmd builds the command line to start Turbostat.
+func (t *Turbostat) buildCmd() []string {
+	cmd := make([]string, 0)
+	if t.UseSudo {
+		cmd = append(cmd, "sudo")
+	}
+	s := int(time.Duration(t.Interval).Seconds())
+	cmd = append(cmd, t.Path, "--quiet", "--interval", strconv.Itoa(s), "--show", "all")
+	return cmd
+}
+
+// processStderr reads error lines from a stream (such as Turbostat stderr)
+// and adds them to an accumulator.
+func processStderr(r io.Reader, acc telegraf.Accumulator) error {
+	scanner := bufio.NewScanner(r)
+	for scanner.Scan() {
+		acc.AddError(errors.New(scanner.Text()))
+	}
+	return scanner.Err()
+}
+
+// processStdout reads metrics from a stream (such as Turbostat stdout)
+// and adds them to an accumulator. If an error is encountered, the function
+// returns it and stops further processing.
+func processStdout(r io.Reader, acc telegraf.Accumulator) error {
+	scanner := bufio.NewScanner(r)
+	if !scanner.Scan() {
+		return scanner.Err()
+	}
+	header := scanner.Text()
+	columns := make([]column, 0)
+	for _, s := range strings.Fields(header) {
+		columns = append(columns, createColumn(s))
+	}
+	for scanner.Scan() {
+		line := scanner.Text()
+		if line == header {
+			continue
+		}
+		values := strings.Fields(line)
+		err := processValues(columns, values, acc)
+		if err != nil {
+			return err
+		}
+	}
+	return scanner.Err()
+}
+
+// Creates a metric from columns and values, and adds it to an accumulator.
+func processValues(columns []column, values []string, acc telegraf.Accumulator) error {
+	if len(values) > len(columns) {
+		return fmt.Errorf("too many values: %d columns, %d values", len(columns), len(values))
+	}
+	tags := make(tagMap, 0)
+	fields := make(fieldMap, 0)
+	timestamps := make([]time.Time, 0)
```

Comment:
There can be only one timestamp?!

Reply:
I now understand the suggestion and implemented it.